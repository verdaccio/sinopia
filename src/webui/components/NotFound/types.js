/**
 * @prettier
 * @flow
 */

export interface IProps {
  pkg: string;
}
