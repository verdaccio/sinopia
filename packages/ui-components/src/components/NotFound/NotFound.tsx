/* eslint-disable verdaccio/jsx-no-style */
import styled from '@emotion/styled';
import FolderOffIcon from '@mui/icons-material/FolderOff';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import React, { useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

import { Theme } from '../../';
import Heading from '../Heading';

const NotFound: React.FC = () => {
  const history = useHistory();
  const { t } = useTranslation();

  const handleGoHome = useCallback(() => {
    history.push('/');
  }, [history]);

  return (
    <Box
      alignItems="center"
      data-testid="404"
      display="flex"
      flexDirection="column"
      flexGrow={1}
      justifyContent="center"
      p={2}
    >
      <Container>
        <FolderOffIcon color="primary" style={{ fontSize: 236 }} />
      </Container>
      <StyledHeading className="not-found-text" variant="h4">
        {t('error.404.sorry-we-could-not-find-it')}
      </StyledHeading>
      <Button data-testid="not-found-go-to-home-button" onClick={handleGoHome} variant="contained">
        {t('button.go-to-the-home-page')}
      </Button>
    </Box>
  );
};

export default NotFound;

const Container = styled('div')({
  margin: '0 auto',
});

const StyledHeading = styled(Heading)<{ theme?: Theme }>(({ theme }) => ({
  color: theme?.palette.mode === 'light' ? theme?.palette.primary.main : theme?.palette.white,
  marginBottom: 16,
}));
