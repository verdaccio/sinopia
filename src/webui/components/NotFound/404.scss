@import '../../styles/variables';
@import '../../styles/mixins';

.notFound {
  width: 100%;
  font-size: $font-size-md;
  line-height: $line-height-xl;
  border: none;
  outline: none;
  @include border-bottom-default($grey-light);

  &:focus {
    @include border-bottom-default($grey);
  }
}
