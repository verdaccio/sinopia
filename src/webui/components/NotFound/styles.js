/**
 * @prettier
 * @flow
 */

import styled from 'react-emotion';

import { fontSize, lineHeight } from '../../utils/styles/sizes';

export const Wrapper = styled.div`
  && {
    font-size: ${fontSize.md};
    line-height: ${lineHeight.xl};
    border: none;
    outline: none;
    flex-direction: column;
    align-items: center;
  }
`;
