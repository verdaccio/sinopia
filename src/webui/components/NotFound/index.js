/**
 * @prettier
 * @flow
 */

import React from 'react';
import { Wrapper } from './styles';
import { IProps } from './types';

const NotFound = ({ pkg }: IProps) => (
  <Wrapper>
    <h1>Error 404 - {pkg}</h1>
    <hr />
    <p>Oops, The package you are trying to access does not exist.</p>
  </Wrapper>
);

export default NotFound;
