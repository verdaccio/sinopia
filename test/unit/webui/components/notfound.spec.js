/**
 * NotFound component
 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import NotFound from '../../../../src/webui/components/NotFound/index';

console.error = jest.fn();

describe('<NotFound /> component', () => {
  it('should load the component in default state', () => {
    const wrapper = mount(<NotFound pkg='test' />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should set html from props', () => {
    const props = {
      pkg: 'verdaccio'
    };
    const wrapper = shallow(<NotFound {...props} />);
    expect(wrapper.find('h1').text()).toEqual('Error 404 - verdaccio');
    expect(wrapper.find('p').text()).toEqual(
      'Oops, The package you are trying to access does not exist.'
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
