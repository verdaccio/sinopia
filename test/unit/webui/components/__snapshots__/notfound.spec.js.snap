// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> component should load the component in default state 1`] = `"<div class=\\"css-24p7uw e1gvymku0\\"><h1>Error 404 - test</h1><hr><p>Oops, The package you are trying to access does not exist.</p></div>"`;

exports[`<NotFound /> component should set html from props 1`] = `"<div class=\\"css-24p7uw e1gvymku0\\"><h1>Error 404 - verdaccio</h1><hr/><p>Oops, The package you are trying to access does not exist.</p></div>"`;
